src/epoch.rs

```diff
@@ -2226,6 +2242,218 @@ impl Epoch {
         me
     }
 
+    /// Converts this epoch into the time of week, represented as a rolling week counter into that time scale and the number of nanoseconds since closest Sunday midnight into that week.
+    /// This is usually how GNSS receivers describe a timestamp.
+    pub fn to_time_of_week(&self) -> (u32, u64) {
+        // fractional days in this time scale
+        let days = self.to_duration().to_unit(Unit::Day);
+        // wk: rolling week counter into timescale
+        let wk = (days / Weekday::DAYS_PER_WEEK).floor() as u32;
+        // tow: number of nanoseconds between self and previous sunday midnight / start of week
+        let mut start_of_week = self.previous_weekday_at_midnight(Weekday::Sunday);
+        // restrict start of week/sunday to the time scale starting day
+        // std::cmp::min( start_of_week, self.time_scale.ref_epoch)
+        match self.time_scale {
```

Comment:
@ChristopherRabotin,

the current implementation behaves funny for extreme values, i'm interested in to_time_of_week for Keplerian equations solving, but it's actually all `Epoch::previous` and `Epoch::next` that are impacted. For these two, extreme values are simply within the first week of a given timescale.

`previous` or `next` have no constraint mechanism to make sure the returned Sunday is not prior  `REF_EPOCH`. I think it is something we should implement, and my current tweak should be moved to these methods, so all calculations requiring weekday operations also benefit from it.

The method i'm interested in is just a use case of it. I decided to return `u16`  but that is arbitrary, i did this because I know that is how GPS vehicles encode such information, so the end user does not have any cast operation to perform. But that is not the point here, all we need to target is correct arithmetic